src/js/form-builder.js

```diff
@@ -1316,34 +1317,32 @@ function FormBuilder(opts, element, $) {
         const senderIsControlsBox = $(ui.sender).attr('id') === $cbUL.attr('id')
 
         const droppingToNewRow = $(ui.item).parent().hasClass(tmpRowPlaceholderClass)
-        const droppingToPlaceholderRow = $(ui.item).parent().hasClass(tmpRowPlaceholderClass)
-        const droppingToExistingRow =
-          $(ui.item).parent().hasClass(rowWrapperClass) && !$(ui.item).parent().hasClass(tmpRowPlaceholderClass)
+        const droppingToExistingRow = !droppingToNewRow && $(ui.item).parent().hasClass(rowWrapperClass)
 
-        if (droppingToNewRow && !senderIsControlsBox) {
-          const colWrapper = $(ui.item)
-
-          const columnData = prepareFieldRow({})
+        if (droppingToNewRow) {
+          if (senderIsControlsBox) {
+            insertTargetIsRow = true
+            insertingNewControl = true
+            $targetInsertWrapper = $(ui.item).parent()
+          } else {
+            const colWrapper = $(ui.item)
 
-          const rowWrapperNode = m('div', null, {
-            id: `${colWrapper.find('li').attr('id')}-row`,
-            className: `row row-${columnData.rowUniqueId} ${rowWrapperClass}`,
-          })
+            const columnData = prepareFieldRow({})
 
-          $(ui.item).parent().replaceWith(rowWrapperNode)
-          AttachColWrapperHandler($(ui.item))
+            const rowWrapperNode = m('div', null, {
+              id: `${colWrapper.find('li').attr('id')}-row`,
+              className: `row row-${columnData.rowUniqueId} ${rowWrapperClass}`,
+            })
 
-          colWrapper.appendTo(rowWrapperNode)
+            $(ui.item).parent().replaceWith(rowWrapperNode)
+            AttachColWrapperHandler($(ui.item))
 
-          setupSortableRowWrapper(rowWrapperNode)
-          syncFieldWithNewRow(colWrapper.attr('id'))
-          checkRowCleanup()
-        }
+            colWrapper.appendTo(rowWrapperNode)
 
-        if (droppingToPlaceholderRow && senderIsControlsBox) {
-          insertTargetIsRow = true
-          insertingNewControl = true
-          $targetInsertWrapper = $(ui.item).parent()
+            setupSortableRowWrapper(rowWrapperNode)
+            syncFieldWithNewRow(colWrapper.attr('id'))
+            checkRowCleanup()
```

Comment:
do we need this if its called below?

Reply:
You are correct. I've removed the redundant checkRowCleanup() here